File: easy15.js
// Solution 1
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    const unique = Array.from(new Set(nums));
    const len = unique.length;
    for (let i = 0; i < len; i++) {
        nums[i] = unique[i];
    }
    
    return len;
};

// Solution 2
var removeDuplicates = function(nums) {
   let k = 1;
   for(let i = 1; nums.length> i; i++) {
      if(nums[i] !== nums[i-1]) {
       nums[k] = nums[i]
       k++
      }
   }
   return k
};

File: easy15.c
// Solution 1
int removeDuplicates(int* nums, int numsSize) {
    int counter = 1;
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[i - 1]) {
            nums[counter] = nums[i];
            counter++;
        }
    }

    return counter;
}

// Solution 2
int removeDuplicates(int* nums, int numsSize) {
    int offset = 0;
    int last = nums[0];

    for (int i = 1; i < numsSize; i++) {
        nums[i - offset] = nums[i];

        if (nums[i] == last) {
            offset++;
        }

        last = nums[i];
    }

    return numsSize - offset;
}
